stl/inc/concepts

```diff
@@ -138,12 +138,14 @@ namespace ranges {
                 noexcept(noexcept(operator()(__t[0], __u[0])))
                 requires requires(_Cpo __fn) { __fn(__t[0], __u[0]); }
             {
+#ifndef __clang__

```

Comment:
To my knowledge, we don't attempt to be strict-aliasing-clean, and we expect Clang to act as a drop-in replacement for MSVC. I don't think we should add these conditional codepaths.

https://clang.llvm.org/docs/UsersManual.html#strict-aliasing says:

> `clang-cl` defaults to `-fno-strict-aliasing`

If we wanted to be globally strict-aliasing-clean, then I'd want some way to validate it at compile-time, and I'd want to behave uniformly for all compilers. (Apparently Clang has an experimental TypeSanitizer so the machinery isn't really available yet?)

Reply:
I roughly remember that we have a general rule "don't help the compiler". Is it dominated by behavior consistency?